### DISCUSSION TITLE
Empty host name error

### DISCUSSION THREAD

**Discussion by Elepert** [keycloak/keycloak#21519] [Q&A]
I'm trying to use Keycloack as a SAML server. I deployed my keycloack (v21.1.2) instance with:
```
docker run --name keycloak_dev2 -p 8089:8080 \                                                 
       -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin \
        quay.io/keycloak/keycloak:latest \
        start-dev --hostname-url=http://localhost:8089
```

I set up a keycloack client with type SAML and various configs that seem fairly basic (can share more if that's useful).

I'm able to get my Python application to redirect to keycloack, am able to login, but in the process of getting redirected back to my application, I get a 500 and the keycloack server has this error:
```
Uncaught server error: java.lang.RuntimeException: empty host name
```

The call that triggers this is:
```http://localhost:8089/realms/<CLIENT_NAME>/protocol/saml?SAMLRequest=<LONG_KEY_HERE>&RelayState=<APPLICATION_URL>```

I've tried specifying all kinds of options to the docker run command (KC_HOSTNAME, --hostname, etc...) and I still get the same error. I can't find anything online about why this error is showing up. Does anyone have any ideas?

**Comment by christian-2** [comment]
Do you implement your own SAML SP in Python (or do you use the [SAML Python Toolkit,](https://github.com/SAML-Toolkits/python-saml) say)? And more specifically, what redirect URIs and Assertion Consumer URLs have you specified in your Keycloak client (of type SAML)?

**Reply by Elepert** [reply]
I do implement my own SP in Python with pysaml. My redirect URIs are here:
<img width="492" alt="Screenshot 2023-07-10 at 09 14 23" src="https://github.com/keycloak/keycloak/assets/23346615/399f4239-8bc3-452e-9366-553e6a8725dd">

I haven't specified any assertion consumer URLs (not even sure where that would be set). 

Also, randomly I restarted my client and the keycloak instance and it started working. I have no idea what I did, I don't think I changed anything.

**Comment by christian-2** [comment]
FYI assertion consumer URLs could be set under _&lt;client&gt;_ / _Advanced_. If I were in your situation, I'd probably want to understand in more detail what is going on at the security perimeter (i.e. with SAML); but circumstances vary, of course.
